index.js

```diff
@@ -267,49 +232,69 @@ function loadUsers() {
     xhr.send();
 }
 
-function loadNotes(user, era, start, end, info) {
+function loadNotes(user, era) {
     // index.html and index.js are loaded through rawgit.com, which routes them
     // through a CDN that caches aggressively. That doesn't work for the data
     // files, which are expected to be updated frequently. So we go through
     // github directly for those -- but note, not the /raw/ URL, since that
     // does not allow CORS, but the raw.githubusercontent.com link.
 
-    const xhr = new XMLHttpRequest();
-    xhr.responseType = 'text';
-    xhr.addEventListener('load', ev => dataLoaded(xhr, era, user, start, end, info));
-    xhr.open("GET", urls.data(user, era));
-    return xhr;
+    return new Promise(resolve => {
+        const xhr = new XMLHttpRequest();
+        xhr.responseType = 'text';
+        xhr.addEventListener('load', ev => resolve(xhr));
+        xhr.open("GET", urls.data(user, era));
+        xhr.send();
+    });
 }
 
 function computeEra(time_sec) {
     return time_sec - (time_sec % ERA_SECONDS);
 }
 
-function loadUserNotes(user, start, end) {
+async function loadUserNotes(user, start, end) {
     if (!end) {
         end = Date.now() / 1000; // ms -> sec
     }
     if (!start) {
         start = end - LOOKBACK_SECONDS;
     }
 
-    let total = 0;
-    for (let t = computeEra(start); t <= computeEra(end); t += ERA_SECONDS) {
-        total++;
-    }
-
-    const info = {
-        total,
-        sofar: 0,
-        found: 0,
-        results: []
-    };
+    const results = [];
+    let found = 0;
```

Comment:
It seems `found` could just be a boolean now.

Reply:
While parallelizing the request, removed this and added the code to check the number of results.